aio_pika/abc.py

```diff
@@ -114,20 +115,36 @@ async def __aexit__(
         raise NotImplementedError
 
 
-HeadersValue = Union[aiormq.abc.FieldValue, bytes]
-HeadersPythonValues = Union[
-    HeadersValue,
-    Set[HeadersValue],
-    Tuple[HeadersValue, ...],
-    FrozenSet[HeadersValue],
-]
-HeadersType = MutableMapping[str, HeadersPythonValues]
+HeadersType = Dict[str, FieldValue]
+
+
+class MessageInfo(TypedDict, total=False):
```

Comment:
Why not dataclass?

Reply:
dataclass with total=False is a very strange solution. Message info it's a way for print message to logs etc. The dictionary is a completely relevant for this.